src/main/java/gov/usgs/ngwmn/logic/WaterLevelStatistics.java

```diff
@@ -173,19 +144,67 @@ public JsonData calculate(Specifier spec, List<WLSample> samples) {
-		if (samples == null || samples.size() == 0) {
+
+	protected void overallLastestPercentile(List<WLSample> samplesByDate) {
+		// get the latest (most recent) sample
+		int last = samplesByDate.size()-1;
+		WLSample latestSample = samplesByDate.get(last);
+		// get the sample for the same month as the latest sample
+		String month = Value.padMonth(Value.monthUTC(latestSample.time));
+		List<WLSample> monthSamples = monthlyStats.filterValuesByGivenMonth(samplesByDate, month);
+		// get the medians for each year-month
+		List<WLSample> normalizeMutlipleYearlyValues = 
+				monthlyStats.medianMonthlyValues(monthSamples,  monthlyStats.sortFunctionByQualifier());
+		// the most recent must now be added into the collection and replace of the year-month it represents
+		replaceLatestSample(normalizeMutlipleYearlyValues, latestSample);
+		// get the percentile of the latest sample
+		BigDecimal latestPercentile = percentileOfValue(normalizeMutlipleYearlyValues, latestSample, Value::valueOf);
+		builder.latestPercentile(latestPercentile.toString());
+	}
+
+	protected void replaceLatestSample(List<WLSample> normalizeMutlipleYearlyValues, WLSample latestSample) {
+		if (normalizeMutlipleYearlyValues.contains(latestSample)) {
+			return; // if it happens to be in there leave it be
+		}
+		// initial conditions and values
+		int        compareToDir  = MediationType.BelowLand.equal(builder.mediation()) ?1 :-1;
+		int        elements      = normalizeMutlipleYearlyValues.size();
+		String     latestYrMonth = latestSample.getTime().substring(0, 7);
+		BigDecimal latestValue   = latestSample.getValue();
+		boolean    isInserting   = true;
+		boolean    isRemoving    = true;
+		// search for the removing current month and the inserting location for latest sample
+		for (int s=0; s<elements && (isRemoving || isInserting); s++) {
+			String yearMonth = ((Value)normalizeMutlipleYearlyValues.get(s)).getTime().substring(0, 7);
+			BigDecimal value = ((Value)normalizeMutlipleYearlyValues.get(s)).getValue();
+			// the data is already sorted, if we exceed the value in the comapreTo direction then insert
+			if (isInserting && latestValue.compareTo(value) == compareToDir) {
+				normalizeMutlipleYearlyValues.add(s, latestSample);
+				elements++; s++;
+				isInserting = false;
+			}
+			// if year-month of the latest sample is found then remove it
+			if (yearMonth.equals(latestYrMonth)) {
+				normalizeMutlipleYearlyValues.remove(s);
+				elements--; s--;
+				isRemoving = false;
+			}
+		}
+		if (isInserting) {
+			normalizeMutlipleYearlyValues.add(latestSample);
+		}
+	}	
+	protected void overallStats(List<WLSample> samplesByDate, List<WLSample> sortedByValue) {
+		if (samplesByDate == null || samplesByDate.size() == 0) {
 			builder.recordYears("0");
 			builder.sampleCount(0);
 			builder.collect();
 			return;
 		}
+		overallLastestPercentile(samplesByDate);
+		overallStatistics.overallStats(samplesByDate, sortedByValue);
```

Comment:
The latest percentile is a bit more nuanced than we originally anticipated. I tried to comment each action explain the decisions.